packages/react-google-maps-api/src/useLoadScript.tsx

```diff
@@ -72,10 +84,12 @@ export function useLoadScript({
       return
     }
 
+    resetStateIfMounted();
```

Comment:
Please separate this logic into its own useEffect. No need to reset when id changes. Also, it's better for clarity of intention.

Reply:
Actually, if id change and url does not change, injectScript will always insert a new script (which imho is not so good because the older script remains, we should provide a strategy to update the id of the existing script instead?)

So basically, I really want to call this resetState whenever injectScript() is called

Reply:
Frankly, I never understood why we even allow customizing the ID considering, that all will always be set to `window.google`. There is no point in having multiple configurations as it will always overwrite the previous one.

Reply:
@FredyC 
id is required to find `<script />` element in DOM to track if script mounted or not

Reply:
Yea sure, but it can be static, right? There is no reason to keep it configurable and a source of confusion. Or at least consider it only initially and don't worry about it when it changes.

Reply:
Agree, I don't see an usecase for having that id change over time. Used this only as a workaround to trigger script updates when injectScript did not reinject on url changes (fixed by my merged PR).

We should rather error if user tries to modify id dynamically imho, because allowing changing id just makes things a bit more complicated for nothing

Reply:
Also, if we could find a way to ensure that at most one hook is mounted, that would ensure more guarantees as well. The user should rather use the hook upper in his tree and only at most once, instead of colocating the load into multiple pages.

For nested/local spinners, one can propagate isLoaded through context

Reply:
Mostly agree, just loader shouldn't be a single one on top because in bigger apps you don't want to be loading map resources before you know they are really needed. Especially on mobile connections.

It could be solved if we add `enabled` to options and then it's possible to drive that through an app state and leave it disabled until really needed.

Reply:
on my Gatsby SPA I just put the hook at the page level, not main app root comp. The load will not happen until the user navigates to a page using maps.

Also, I handle cleanup (mostly to avoid the Gmaps error due to loading twice)

```
const useGoogleMapsAPILoader = () => {
  const language = useGoogleMapsLanguage();
  const scriptId = `GoogleMaps_lang=${language}`; // Should not be needed anymore now that injectScript is fixed
  useEffect(
    () => {
      return () => {
        const script = document.getElementById(scriptId);
        if (script) {
          script.remove();
          // @ts-ignore
          if (window.google && window.google.maps) {
            // @ts-ignore
            delete window.google.maps;
          }
        }
      };
    },
    [language],
  );

  return useLoadScript({
    id: scriptId,
    language,
    libraries: AppMapLibraries,
    googleMapsApiKey: AppConfiguration.GoogleMapsApiKey,
  });
}; 
```

Reply:
it works fine, the only thing that annoys me with that system is that if maps has been loaded on page1, and user goes to page 2, then the map will always reload even if the url is the same. So a solution to enable/disable at the root of my app would be useful ;)